src/main/scala/com/itechart/ny_accidents/Application.scala

```diff
@@ -2,41 +2,43 @@ package com.itechart.ny_accidents
 
 import com.itechart.ny_accidents.constants.Configuration
 import com.itechart.ny_accidents.constants.Injector.injector
-import com.itechart.ny_accidents.database.NYDataDatabase
-import com.itechart.ny_accidents.database.dao.PopulationStorage
+import com.itechart.ny_accidents.database.DistrictsStorage
 import com.itechart.ny_accidents.database.dao.cache.MergedDataCacheDAO
-import com.itechart.ny_accidents.entity.{Accident, MergedData}
+import com.itechart.ny_accidents.database.dao.{DistrictsDAO, PopulationStorage}
+import com.itechart.ny_accidents.entity._
 import com.itechart.ny_accidents.parse.AccidentsParser
 import com.itechart.ny_accidents.report.Reports
 import com.itechart.ny_accidents.report.generators._
-import com.itechart.ny_accidents.service.MergeService
 import com.itechart.ny_accidents.service.metric.{DayPeriodMetricService, PopulationMetricService, WeatherMetricService}
-import org.apache.spark.rdd.RDD
+import com.itechart.ny_accidents.service.{DistrictsService, MergeService, WeatherMappingService}
+import com.itechart.ny_accidents.utils.FileWriterUtils
+import org.apache.spark.sql.Dataset
 import org.slf4j.LoggerFactory
 
+
 object Application extends App {
   lazy val logger = LoggerFactory.getLogger(getClass)
 
-  val accidentsParser = injector.getInstance(classOf[AccidentsParser])
-  val mergeService = injector.getInstance(classOf[MergeService])
   val weatherMetricService = WeatherMetricService
   val dayPeriodMetricService = DayPeriodMetricService
   val cacheService = injector.getInstance(classOf[MergedDataCacheDAO])
   val populationService = injector.getInstance(classOf[PopulationMetricService])
   val populationStorage = injector.getInstance(classOf[PopulationStorage])
-  sys.addShutdownHook(cacheService.close)
+  val weatherMappingService = injector.getInstance(classOf[WeatherMappingService])
+  val districtService = injector.getInstance(classOf[DistrictsService])
+  val districtDao = injector.getInstance(classOf[DistrictsDAO])
+  val districtsStorage = injector.getInstance(classOf[DistrictsStorage])
+  val reports = injector.getInstance(classOf[Reports])
 
-  val rawData = accidentsParser.readData(Configuration.DATA_FILE_PATH).cache()
-  logger.info("Raw data read")
 
-  val mergeData: RDD[MergedData] = mergeService
-    .mergeAccidentsWithWeatherAndDistricts[Accident, MergedData](rawData, mergeService.fullMergeMapper).cache()
+  val accidents: Dataset[AccidentSparkFormat] = AccidentsParser.readData(Configuration.DATA_FILE_PATH).cache()
+  logger.info("Raw data read - " + accidents.count())
+  val mergedData = MergeService.mergeData(accidents).cache()
 
-  logger.info("Merged data size: " + mergeData.count())
 
-  val creationDate = org.apache.spark.sql.functions.current_date()
-  val reports = injector.getInstance(classOf[Reports])
+  println("Merged data length: " + mergedData.count)
```

Comment:
What was the reason of replacing logger with printing to console?

Reply:
@abelyaev-itechart 
fixed
e18dbc8a6616d0d03a051a8ef80c3b59a11017ac